src/smt/mod.rs

```diff
@@ -55,12 +72,21 @@ impl<'ctx> SmtCtx<'ctx> {
             for spec in &decl.body {
                 if let DomainSpec::Function(func_ref) = &spec {
                     let func = func_ref.borrow();
-                    let domain: Vec<Sort<'_>> = func
-                        .inputs
-                        .node
-                        .iter()
-                        .map(|param| ty_to_sort(self, &param.ty))
-                        .collect();
+
+                    let mut domain = if self.is_limited_function_decl(&func) {
```

Comment:
Maybe capture this functionality in some method somewhere else in a section/module of code specifically for the fuel transformation?